### PR TITLE
Add ability to reset builds

### PR DESCRIPTION
**Pull request by stuartwdouglas**
By adding the jvmbuildservice.io/rebuild: true annotation the system
knows to attempt to rebuild an ArtifactBuild.

### CODE REVIEW

**Comment by gabemontero** [comment]
all 8 PaC image builds are in fact OK:

```
gmontero ~ $ oc get pr  -n jvm-build-service jvm-build-cache-on-pull-request-cw8ng
NAME                                    SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-cache-on-pull-request-cw8ng   True        Completed   120m        108m
gmontero ~ $ oc get pr -n jvm-build-service jvm-build-controller-on-pull-request-h4chq
NAME                                         SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-controller-on-pull-request-h4chq   True        Completed   120m        111m
gmontero ~ $ oc get pr  -n jvm-build-service jvm-build-hacbs-jdk11-builder-on-pull-request-kbrbg
NAME                                                  SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-hacbs-jdk11-builder-on-pull-request-kbrbg   True        Completed   121m        104m
gmontero ~ $ oc get pr  -n jvm-build-service jvm-build-hacbs-jdk17-builder-on-pull-request-cqm6k
NAME                                                  SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-hacbs-jdk17-builder-on-pull-request-cqm6k   True        Completed   121m        107m
gmontero ~ $ oc get pr  -n jvm-build-service jvm-build-hacbs-jdk8-builder-on-pull-request-cbvx8
NAME                                                 SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-hacbs-jdk8-builder-on-pull-request-cbvx8   True        Completed   122m        104m
gmontero ~ $ oc get pr  -n jvm-build-service jvm-build-request-processor-on-pull-request-xlfhk
NAME                                                SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-request-processor-on-pull-request-xlfhk   True        Completed   122m        110m
gmontero ~ $ oc get pr  -n jvm-build-service jvm-build-sidecar-on-pull-request-6jc84
NAME                                      SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-build-sidecar-on-pull-request-6jc84   True        Completed   123m        111m
gmontero ~ $ oc get pr  -n jvm-build-service jvm-dependency-analyser-on-pull-request-5dmx6
NAME                                            SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
jvm-dependency-analyser-on-pull-request-5dmx6   True        Completed   123m        110m
gmontero ~ $ 
```

File: java-components/adhoc-analyser-tool/src/main/java/com/redhat/hacbs/analyser/kube/ResetArtifactBuildsCommand.java [hunk 1]
```diff
@@ -31,15 +32,15 @@ public void run() {
...
+                request.getMetadata().setAnnotations(Map.of("jvmbuildservice.io/rebuild", "true"));
+                client.createOrReplace(request);
             } else {
                 List<ArtifactBuild> items = client.list().getItems();
                 for (var request : items) {
                     if (!missing || request.getStatus().getState().equals("ArtifactBuildMissing")) {
-                        request.getStatus().setState("");
-                        client.updateStatus(request);
+                        request.getMetadata().setAnnotations(Map.of("jvmbuildservice.io/rebuild", "true"));
+                        client.createOrReplace(request);
```
Comment (gabemontero) [inline java-components/adhoc-analyser-tool/src/main/java/com/redhat/hacbs/analyser/kube/ResetArtifactBuildsCommand.java:43]: understanding this comment is not specifically related to this change, and quite possibly would be addressed in a separate PR, but currently we only give one shot for this api server invocation to pass or fail

intermittent / retryable errors are certainly part of k8s api server interaction - does the underlying client already handle that, or should we start incorporating that throughout the java component use if k8s client ?

File: pkg/reconciler/artifactbuild/artifactbuild.go [hunk 2]
```diff
@@ -454,6 +460,41 @@ func (r *ReconcileArtifactBuild) handleStateBuilding(ctx context.Context, abr *v
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileArtifactBuild) handleRebuild(ctx context.Context, abr *v1alpha1.ArtifactBuild) (reconcile.Result, error) {
```
Comment (gabemontero) [inline pkg/reconciler/artifactbuild/artifactbuild.go:463]: any reason not to do a unit test ?

**Review by gabemontero — COMMENTED** [review]
couple of minor items / possible follow up PR candidates

File: java-components/adhoc-analyser-tool/src/main/java/com/redhat/hacbs/analyser/kube/ResetArtifactBuildsCommand.java (hunk 1 above)
Comment (stuartwdouglas) [reply java-components/adhoc-analyser-tool/src/main/java/com/redhat/hacbs/analyser/kube/ResetArtifactBuildsCommand.java:43]: This is basically just a tool that I am using for local dev to test things out. We don't actually use it in production anywhere.

Comment (gabemontero) [reply java-components/adhoc-analyser-tool/src/main/java/com/redhat/hacbs/analyser/kube/ResetArtifactBuildsCommand.java:43]: ah yeah I think you've told me this before ....I need to try and remember and pay attention to "adhoc" being in the file name and that it means it is non-production use

**Comment by gabemontero** [comment]
ok only the controller image pipeline as acode job is stuck with "in progress" but when I look in the staging cluster:

`jvm-build-controller-on-pull-request-kdgjg   True        Completed   36m         28m`

so let's merge if the e2e's are green

**Pull request merged**